src/lib.rs

```diff
@@ -63,28 +78,28 @@ where
 
     /// Swap a new value into the Atom, This will try multiple
     /// times until it succeeds. The old value will be returned.
-    pub fn swap(&self, v: P, order: Ordering) -> Option<P> {
+    pub fn swap(&self, v: P) -> Option<P> {
         let new = v.into_raw();
-        let old = self.inner.swap(new, order);
+        let old = self.inner.swap(new, Ordering::AcqRel);
         unsafe { Self::inner_from_raw(old) }
     }
 
     /// Take the value of the Atom replacing it with null pointer
     /// Returning the contents. If the contents was a `null` pointer the
     /// result will be `None`.
-    pub fn take(&self, order: Ordering) -> Option<P> {
-        let old = self.inner.swap(ptr::null_mut(), order);
+    pub fn take(&self) -> Option<P> {
+        let old = self.inner.swap(ptr::null_mut(), Ordering::AcqRel);
         unsafe { Self::inner_from_raw(old) }
     }
 
     /// This will do a `CAS` setting the value only if it is NULL
     /// this will return `None` if the value was written,
     /// otherwise a `Some(v)` will be returned, where the value was
     /// the same value that you passed into this function
-    pub fn set_if_none(&self, v: P, order: Ordering) -> Option<P> {
+    pub fn set_if_none(&self, v: P) -> Option<P> {
         let new = v.into_raw();
-        let old = self.inner.compare_and_swap(ptr::null_mut(), new, order);
-        if !old.is_null() {
+        let result = self.inner.compare_exchange(ptr::null_mut(), new, Ordering::AcqRel, Ordering::Acquire);
```

Comment:
The `Acquire` ordering is redundant here

```suggestion
        let result = self.inner.compare_exchange(ptr::null_mut(), new, Ordering::Release, Ordering::Relaxed);
```